experimental/unitsconversion/src/info_provider.rs

```diff
@@ -0,0 +1,160 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::prelude::*;
+use zerovec::{VarZeroVec, ZeroMap};
+
+#[cfg(feature = "compiled_data")]
+#[derive(Debug)]
+/// Baked data
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. In particular, the `DataProvider` implementations are only
+/// guaranteed to match with this version's `*_unstable` providers. Use with caution.
+/// </div>
+pub struct Baked;
+
+#[cfg(feature = "compiled_data")]
+const _: () = {
+    pub mod icu {
+        pub use crate as unitsconversion;
+    }
+    icu_unitsconversion_data::make_provider!(Baked);
+    icu_unitsconversion_data::impl_units_info_v1!(Baked);
+};
+
+#[cfg(feature = "datagen")]
+/// The latest minimum set of keys required by this component.
+pub const KEYS: &[DataKey] = &[UnitsInfoV1Marker::KEY];
+
+/// This type encapsulates all the constant data required for unit conversions.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[icu_provider::data_struct(marker(UnitsInfoV1Marker, "units/info@1", singleton))]
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct UnitsInfoV1<'data> {
+    /// Maps from unit name (e.g. foot) to the index of the unit in the `unit_quantity` vector.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub units_info: ZeroMap<'data, str, UnitsInfoIndex>,
+
+    /// Contains the quantity information for the units.
+    /// For example, the quantity for the unit `foot` is `length`.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub unit_quantity: VarZeroVec<'data, UnitQuantityULE>,
+
+    /// Contains the conversion information, such as the conversion rate and the base unit.
+    /// For example, the conversion information for the unit `foot` is `1 foot = 0.3048 meter`.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub convert_units: VarZeroVec<'data, ConvertUnitsULE>,
```

Comment:
It's unclear to me what we're achieving by having this in a separate VZV instead of in units_info. There shouldn't be any duplication, as two units will never have the same conversion to a third unit.

Reply:
This is the comment I was referring two. There is duplication here, each element of this VZV is used exactly once, so there' no point in having it in a separate vector. This probably increases data size, and definitely adds additional derefs, additional invariants/fallibility, and worsens cache locality.

Reply:
This also makes the JSON hard to understand. I can see that there are three conversions with base `kilogram-meter-per-square-second` in this vector, but I cannot figure out which units they belong to.

Reply:
I have already filed an issue for a follow-up PR: https://github.com/unicode-org/icu4x/issues/4313